dtfs-central-api/src/v1/controllers/portal/deal/get-deal.controller.js

```diff
@@ -52,9 +53,21 @@ const findOneDeal = async (_id, callback) => {
     throw new Error('Invalid Deal Id');
   }
 
-  const dealsCollection = await db.getCollection(MONGO_DB_COLLECTIONS.DEALS);
+  const portalDealsCollection = await db.getCollection(MONGO_DB_COLLECTIONS.DEALS);
+  const tfmDealsCollection = await db.getCollection(MONGO_DB_COLLECTIONS.TFM_DEALS);
 
-  const deal = await dealsCollection.findOne({ _id: { $eq: ObjectId(_id) } });
+  const portalDeal = await portalDealsCollection.findOne({ _id: { $eq: ObjectId(_id) } });
+
+  if (!portalDeal) {
+    if (callback) {
+      callback(null);
+    }
+    return null;
+  }
```

Comment:
I really don't like this callback but Im not sure how to keep this compatible otherwise